rasa/server.py

```diff
@@ -1422,34 +1422,6 @@ def _test_data_file_from_payload(request: Request, temporary_directory: Path) ->
     )
 
 
-def _validate_json_training_payload(rjs: Dict) -> None:
-    if "config" not in rjs:
-        raise ErrorResponse(
-            HTTPStatus.BAD_REQUEST,
-            "BadRequest",
-            "The training request is missing the required key `config`.",
-            {"parameter": "config", "in": "body"},
-        )
-
-    if "nlu" not in rjs and "stories" not in rjs:
-        raise ErrorResponse(
-            HTTPStatus.BAD_REQUEST,
-            "BadRequest",
-            "To train a Rasa model you need to specify at least one type of "
-            "training data. Add `nlu` and/or `stories` to the request.",
-            {"parameters": ["nlu", "stories"], "in": "body"},
-        )
```

Comment:
This is a functionality change, not just docs change - Can you tag an additional reviewer because I am not caught up on why this should or shouldn't happen

Reply:
Ope I see it auto tagged someone, so that's good

Reply:
I'm removing code that's not used anymore. JSON was not supported, this function is not used currently. People have said docs had been stale for some time now.

Reply:
Ah right, I see now there's no references left to this function anywhere 👍